geode-client-native/src/cppcache/impl/TcpConn.hpp

```diff
@@ -41,6 +41,8 @@ namespace gemfire {
     protected:
 
       ACE_INET_Addr m_addr;
+	  //Copyright: Amdocs Software Systems Limited, 2016
+      uint8 m_suffixFiller[32]; // Protection from the ACE bug
```

Comment:
Hi,

Yes –

ACE has memory corruption in the class that appears one line above this filler (ACE_INET_Addr). We found it using Valgrind analyzer with a small testing application that we wrote (all it does is connecting the c++ client to the server), Valgrind showed a corruption in this class, and we didn’t see any problem in the client’s code with using this class, and we didn’t have time to investigate further, but when adding the filler – the process stopped crashing.

Thanks,
Gal


From: Vishal Rao [mailto:notifications@github.com]
Sent: Wednesday, July 13, 2016 14:20
To: apache/incubator-geode
Cc: Gal Palmery; Author
Subject: Re: [apache/incubator-geode] GEODE-88: code fixes for c++ client (#199)


In geode-client-native/src/cppcache/impl/TcpConn.hpp<https://github.com/apache/incubator-geode/pull/199#discussion_r70607295>:

> @@ -41,6 +41,8 @@ namespace gemfire {

>      protected:

>

>        ACE_INET_Addr m_addr;

> +       //Copyright: Amdocs Software Systems Limited, 2016

> +      uint8 m_suffixFiller[32]; // Protection from the ACE bug

Can you tell more about what ACE bug this is guarding against?

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub<https://github.com/apache/incubator-geode/pull/199/files/d70c5f917f3050365b5d4f6f81d38f26c9542481#r70607295>, or mute the thread<https://github.com/notifications/unsubscribe/ARLWfefrECe4o4lZ7SBLDDYxVET-r_0mks5qVMnvgaJpZM4JLJVC>.

This message and the information contained herein is proprietary and confidential and subject to the Amdocs policy statement,
you may review at http://www.amdocs.com/email_disclaimer.asp